internal/info/info.go

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (C) 2019 Vdaas.org Vald team ( kpango, kmrmt, rinx )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package info provides build-time info
+package info
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/vdaas/vald/internal/log"
+)
+
+var (
+	Version   = "v0.0.1"
+	GitCommit = "no commit info available."
+	BuildTime = time.Now().Format(time.RFC1123)
+
+	GoVersion  string
+	GoOS       string
+	GoArch     string
+	CGOEnabled string
+	NGTVersion string
+)
+
+func ShowVersionInfo(extras map[string]string) func(name string) {
+	return func(name string) {
+		defaultKeys := []string{"version", "commit hash", "build time"}
+		keys := make([]string, 0, len(extras))
+		for k := range extras {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+
+		var l int
+		maxlen := 0
+		for _, k := range append(defaultKeys, keys...) {
+			l = len(k)
+			if maxlen < l {
+				maxlen = l
+			}
+		}
+
+		infoFormat := fmt.Sprintf("%%-%ds -> %%s", maxlen)
+
+		log.Infof("vald %s server", name)
+		log.Infof(infoFormat, defaultKeys[0], log.Bold(Version))
+		log.Infof(infoFormat, defaultKeys[1], GitCommit)
+		log.Infof(infoFormat, defaultKeys[2], BuildTime)
```

Comment:
could you please build all information as string variable then execute log.Info once

Reply:
okay. i revised it in 3ba0575 .
it shows server information as below. Is it okay?

```
% docker run -it vdaas/vald-agent-ngt:latest --version
2019-12-28 09:12:26     [INFO]: vald agent ngt server
version     -> v0.0.1
commit hash -> c347ee8f7c1b148019ffb3f22f106ff4622f6c05
build time  -> Sat, 28 Dec 2019 09:12:26 UTC
arch        -> amd64
cgo enabled -> 1
go version  -> 1.13.5
ngt version -> 1.8.4
os          -> linux
```